infra/modules/containerapp.bicep

```diff
@@ -0,0 +1,105 @@
+param location string
+param environmentName string
+param uniqueSuffix string
+param tags object
+param containerImage string
+param openAiEndpoint string
+param documentIntelligenceEndpoint string
+param speechResourceId string
+
+var containerAppName = 'ca-${environmentName}-${uniqueSuffix}'
+var containerEnvName = 'cae-${environmentName}-${uniqueSuffix}'
+var logAnalyticsName = 'log-${environmentName}-${uniqueSuffix}'
+
+resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2022-10-01' = {
+  name: logAnalyticsName
+  location: location
+  tags: tags
+  properties: {
+    sku: {
+      name: 'PerGB2018'
+    }
+    retentionInDays: 30
+    features: {
+      enableLogAccessUsingOnlyResourcePermissions: true
+    }
+  }
+}
+
+resource containerAppEnv 'Microsoft.App/managedEnvironments@2023-05-01' = {
+  name: containerEnvName
+  location: location
+  tags: tags
+  properties: {
+    appLogsConfiguration: {
+      destination: 'log-analytics'
+      logAnalyticsConfiguration: {
+        customerId: logAnalytics.properties.customerId
+        sharedKey: logAnalytics.listKeys().primarySharedKey
+      }
+    }
+  }
+}
+
+resource containerApp 'Microsoft.App/containerApps@2024-10-02-preview' = {
+  name: containerAppName
+  location: location
+  tags: tags
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {
+    managedEnvironmentId: containerAppEnv.id
+    configuration: {
+      activeRevisionsMode: 'Single'
+      ingress: {
+        external: true
+        targetPort: 9000
+        stickySessions: {
+          affinity: 'sticky'
+        }
+      }
+    }
+    template: {
+      containers: [
+        {
+          name: containerAppName
+          image: containerImage
+          env: [
+            {
+              name: 'AZURE_OPENAI_ENDPOINT'
+              value: openAiEndpoint
+            }
+            {
+              name: 'AZURE_OPENAI_MODEL_DEPLOYMENT'
+              value: 'gpt-4o'
```

Comment:
Can we make this a variable (based on the `gpt-4o` name set earlier as model deployment).